File: Data Structures & Algorithms/two-integer-sum/submission-8.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int diff = target - num;
            if (map.get(diff) != null) {
                int j = map.get(diff);
                if (i < j) {
                    return new int[]{i, j};
                } else {
                    return new int[]{j, i};
                }
            }
            map.put(num, i);
        }

        return new int[2];
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-8.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        r, c = 0, len(matrix[0]) - 1
        
        while r < len(matrix) and c >= 0:
            curr = matrix[r][c]

            if curr == target:
                return True
            elif curr < target:
                r += 1
            else:
                c -= 1
        
        return False

        